cyclonedds/idl/_main.py

```diff
@@ -106,21 +111,21 @@ def populate_locked(self):
                 self.member_ids = ids
 
             from ._builder import Builder
-            self.v0_machine, self.v2_machine, self.keyless, self.version_support = Builder.build_machines(self.datatype)
-
-            if self.version_support.SupportsBasic & self.version_support:
-                self.v0_keyresult: KeyScanner = self.v0_machine.key_scan()
-                if self.v0_keyresult.rtype != KeyScanResult.PossiblyInfinite and self.v0_keyresult.size <= 16:
-                    self.v0_key_max_size = self.v0_keyresult.size
-                else:
-                    self.v0_key_max_size = 17  # or bigger ;)
-
-            if self.version_support.SupportsV2 & self.version_support:
-                self.v2_keyresult: KeyScanner = self.v2_machine.key_scan()
-                if self.v2_keyresult.rtype != KeyScanResult.PossiblyInfinite and self.v2_keyresult.size <= 16:
-                    self.v2_key_max_size = self.v2_keyresult.size
-                else:
-                    self.v2_key_max_size = 17  # or bigger ;)
+            self.v1_machine, self.v2_machine, self.data_type_props, self.supported_versions, self.default_version = Builder.build_machines(self.datatype)
+            # FIXME: move DATA_TYPE_CONTAINS_... somewhere where we can access it
```

Comment:
Maybe move the `DDS_DATA_TYPE_CONTAINS_` defines to `cyclonedds/idl/_support.py`?